### ISSUE TITLE
Calling agent.createVerifiableCredential() nests the field credentialSchema into newly created field vc

### ISSUE THREAD

**Issue by plesastapevka** [uport-project/veramo#820]
**Bug severity**
3

**Describe the bug**
When creating and issuing a VC with the method `agent.createVerifiableCredential()` as seen in the code below, the custom added field `credentialSchema` gets nested into the field `vc`. See the steps to reproduce. As shown in the [Example 18 at W3C Verifiable Credential Data Model Documentation](https://www.w3.org/TR/vc-data-model/#example-usage-of-the-credentialschema-property-to-perform-json-schema-validation), the `credentialSchema` field should be top-level.

**To Reproduce**
1. Create a Verifiable Credential
    ```javascript
    const verifiableCredential = await agent.createVerifiableCredential({
      credential: {
        issuer: {
          id: "did:ethr:example",
        },
        "@context": [
          "https://www.w3.org/2018/credentials/v1",
          "https://beta.api.schemas.serto.id/v1/public/program-completion-certificate/1.0/ld-context.json",
        ],
        // Note this field here
        credentialSchema: {
          id: "https://beta.api.schemas.serto.id/v1/public/program-completion-certificate/1.0/json-schema.json",
          type: "JsonSchemaValidator2018",
        },
        type: ["VerifiableCredential", "ProgramCompletionCertificate"],
        issuanceDate: new Date().toISOString(),
        credentialSubject: {
          id: "did:ethr:examplesubject",
          accomplishmentType: "Example Type",
          learnerName: "Example Learner",
          achievement: "Example Achievement",
          courseProvider: "Example Provider",
          },
       },
       save: false,
       proofFormat: "jwt",
     });
    ```

**Observed behaviour**
Actual returned Verifiable Credential:
```
{
  // Note this new field vc and credentialSchema inside of it
  vc: {
    credentialSchema: {
      id: 'https://beta.api.schemas.serto.id/v1/public/program-completion-certificate/1.0/json-schema.json',
      type: 'JsonSchemaValidator2018'
    }
  },
  credentialSubject: {
    accomplishmentType: 'Example Type',
    learnerName: 'Example Learner',
    achievement: 'Example Achievement',
    courseProvider: 'Example Provider',
    id: 'did:ethr:examplesubject'
  },
  issuer: {
    id: 'did:ethr:example'
  },
  type: [ 'VerifiableCredential', 'ProgramCompletionCertificate' ],
  '@context': [
    'https://www.w3.org/2018/credentials/v1',
    'https://beta.api.schemas.serto.id/v1/public/program-completion-certificate/1.0/ld-context.json'
  ],
  issuanceDate: 'date in ISO format',
  proof: {
    type: 'JwtProof2020',
    jwt: 'jwt'
  }
}
```
   
**Expected behaviour**
Expected returned Verifiable Credential as per mentioned [W3C Example 18](https://www.w3.org/TR/vc-data-model/#example-usage-of-the-credentialschema-property-to-perform-json-schema-validation):

```
{
  // Note this field in top-level as it should be
  "credentialSchema": {
    "id": "https://beta.api.schemas.serto.id/v1/public/program-completion-certificate/1.0/json-schema.json",
    "type": "JsonSchemaValidator2018"
  },
  "credentialSubject": {
    "accomplishmentType": "Example Type",
    "learnerName": "Example Learner",
    "achievement": "Example Achievement",
    "courseProvider": "Example Provider",
    "id": "did:ethr:examplesubject"
  },
  "issuer": {
    "id": "did:ethr:example"
  },
  "type": [ "VerifiableCredential", "ProgramCompletionCertificate" ],
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://beta.api.schemas.serto.id/v1/public/program-completion-certificate/1.0/ld-context.json"
  ],
  "issuanceDate": "date in ISO format",
  "proof": {
    "type": "JwtProof2020",
    "jwt": "jwt"
  }
}

```

**Versions (please complete the following information):**
 - Veramo: v3.1.0
 - Browser: /
 - Node Version v14.15.0

**Comment by mirceanis** [comment]
The root cause of this bug is in [`did-jwt-vc`](https://github.com/decentralized-identity/did-jwt-vc) which is the library we use to parse JWT credentials.

Can you also post a sample JWT (if it safe to post it in public)?
The JWT payload may not be malformed, only the parsed variant.

**Comment by plesastapevka** [comment]
I think everything is good with `veramo` and `did-jwt-vc`. After a long time of research and trial and error approach, we found out that there might be some misunderstanding with `veramo` and `serto-tools-vc` interoperability, so there's no actual bug regarding veramo. I'm closing the issue now.

**Comment by mirceanis** [comment]
Would you like to share the misunderstanding here?
I think it's best to correct it if possible :)

**Comment by plesastapevka** [comment]
Sorry for mega late answer, I've been really busy and totally forgot about this issue!

The problem was our desire to combine `veramo` and `serto-tools-vc` libraries when `serto-tools-vc` isn't (or at least wasn't at the time of opening this issue) fully up-to-date with all the standards I think. When validating a Verifiable Credential with `serto-tools-vc` there was a different expected structure of VC than what we got from issuing with `veramo`. For our purposes we omitted the `serto-tools-vc` library and went on with validating the VC explicitly with `veramo` and to check if the schema is correct we used [`jsonschema`](https://github.com/tdegrunt/jsonschema#readme) (our VC validation is still in development so I cannot assure you that this lib is 100% working).

So proposed solution for our use case was:
1. First validate the issued VC with `veramo` 
2. Fetch VC JSON Schema from [Serto Schemas](https://schemas.serto.id/schemas) ([i.e. Program Completion Course](https://beta.api.schemas.serto.id/v1/public/program-completion-certificate/1.0/json-schema.json)) - We're also looking for and researching better decentralized solutions for storing schemas.
3. Verify that the issued VC conforms the fetched VC Schema, using [`jsonschema`](https://github.com/tdegrunt/jsonschema#readme).
